### DISCUSSION TITLE
How C++ application can be compiled with EDK2?

### DISCUSSION THREAD

**Discussion by atomer01** [tianocore/edk2#2765] [Q&A]
We have C++ application and we want to enable that for UEFI as well.
How application can be compiled with EDK2?

**Comment by spbrogan** [comment]
I am not aware of c++  being a feature of the tianocore/edk2 build system. 
If you want to build an application for execution in the UEFI environment there are some samples located here: 
https://github.com/tianocore/edk2/tree/master/MdeModulePkg/Application

**Comment by namazso** [comment]
C++ was probably intended at some point (since it's in the build rules) but is broken by default. If you add the ".cpp" extension here: https://github.com/tianocore/edk2/blob/master/BaseTools/Source/Python/AutoGen/GenMake.py#L1113 it should be able to compile .cpp files at least. Obviously you would still be unable to use STL features, but C with templates and classes is already better
